test/functional/feature_swiftsync.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+""" Testing related to SwiftSync initial block download protocol.
+
+This test creates a node that acts as a single source of truth for two other nodes.
+It tests three properties:
+- A correctly constructed hints file allows a new client to perform initial block download
+- Block reorganizations are handled by the new node online
+- A faulty hints file is recognized by a new client and the new client can recover with a `-reindex`
+"""
+from test_framework.util import assert_equal
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.wallet import MiniWallet
+
+BASE_BLOCKS = 200
+NUM_SWIFTSYNC_BLOCKS = 100
+NUM_ADDITIONAL_BLOCKS = 100
+GOOD_FILE = "good.hints"
+BAD_FILE = "bad.hints"
+
+class SwiftSyncTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 3
+        self.setup_clean_chain = True
+
+    def setup_network(self):
+        self.add_nodes(3)
+
+    def run_test(self):
+        full_node = self.nodes[0]
+        sync_node = self.nodes[1]
+        fail_node = self.nodes[2]
+        self.start_node(0)
+        mini_wallet = MiniWallet(full_node)
+        ## Coinbase outputs are treated differently by the SwiftSync protocol as their inputs are ignored.
+        ## To ensure the hash aggregate is working correctly, we also create non-coinbase transactions.
+        self.log.info(f"Generating {BASE_BLOCKS} blocks to a source node")
+        self.generate(mini_wallet, BASE_BLOCKS, sync_fun=self.no_op)
+        self.log.info(f"Sending {NUM_SWIFTSYNC_BLOCKS} self transfers")
+        for _ in range(NUM_SWIFTSYNC_BLOCKS):
+            mini_wallet.send_self_transfer(from_node=full_node)
+            self.generate(full_node, nblocks=1, sync_fun=self.no_op)
+        self.log.info("Creating hints file")
+        result = full_node.generatetxohints(GOOD_FILE)
+        hints_path = result["path"]
+        self.log.info(f"Created hints file at {hints_path}")
+        assert_equal(full_node.getblockcount(), NUM_SWIFTSYNC_BLOCKS + BASE_BLOCKS)
```

Comment:
86d009d0009a53bbb42923b1c1c062f11499d481: You could also assert that the block count is the same as `result["height"]`.